src/lib.rs

```diff
@@ -176,3 +196,153 @@ fn no_lazy_static_rule() -> Rule {
         .build()
         .unwrap()
 }
+
+#[macro_export]
+macro_rules! assert_node_kind {
+    ($node:expr, $kind:literal $(,)?) => {{
+        assert_eq!($node.kind(), $kind);
+        $node
+    }};
+}
+
+fn get_constrained_type_parameter_name(node: Node) -> Node {
+    assert_node_kind!(node, "constrained_type_parameter");
+    let name_node = node.child_by_field_name("left").unwrap();
+    assert_node_kind!(name_node, "type_identifier");
+    name_node
+}
+
+fn maybe_get_ancestor_node_of_kind<'node>(
+    mut node: Node<'node>,
+    kind: &str,
+) -> Option<Node<'node>> {
+    while node.kind() != kind {
+        match node.parent() {
```

Comment:
This looks like I could write it as `node = node.parent()?;`

Reply:
Ah yeah, I don't think I've ever happened to write a `while` like this inside a function returning an `Option`, but that makes sense.

Reply:
Ok updated on current branch